src/DatePicker/DatePicker.web.js

```diff
@@ -1,52 +1,188 @@
 // @flow
 
 import * as React from 'react';
-import DatePicker from 'react-datepicker';
-import 'react-datepicker/dist/react-datepicker.css';
+import { View, TouchableWithoutFeedback } from 'react-native';
+import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+import { StyleSheet } from '../PlatformStyleSheet';
+import DatePickerDayTile from './DatePickerDayTile';
+import { Button } from '../Button';
+import { Text } from '../Text';
 
 import type { Props } from './DatePickerTypes';
 
 type State = {
   date: Date,
+  month: number,
+  year: number,
+};
+
+const parsePropsToState = date => {
+  const tempDate = date ?? new Date();
+  return {
+    date: tempDate,
+    month: tempDate.getMonth(),
+    year: tempDate.getFullYear(),
+  };
 };
 
 export default class WebDatePicker extends React.Component<Props, State> {
   constructor(props: Props) {
     super(props);
 
-    this.state = {
-      date: props.date || new Date(),
-    };
+    this.state = parsePropsToState(props.date);
   }
 
-  handleConfirm = (date: Date) => {
+  componentDidUpdate = (prevProps: Props) => {
+    const { date } = this.props;
+    if (date && date !== prevProps.date) {
+      this.setState(parsePropsToState(date));
+    }
+  };
+
+  handleDaySelect = (dayId: number) => {
     const { onConfirm } = this.props;
+    const { month, year } = this.state;
+    const date = new Date(year, month, dayId);
+    this.setState(parsePropsToState(date));
+    onConfirm(date);
+  };
 
-    this.setState({
-      date,
+  swapMonthBack = () => {
+    this.setState(state => {
+      const month = state.month === 0 ? 11 : state.month - 1;
+      return {
+        month,
+        year: month === 11 ? state.year - 1 : state.year,
+      };
     });
+  };
 
-    onConfirm(date);
+  swapMonthForward = () => {
+    this.setState(state => {
+      const month = (state.month + 1) % 12;
+      return {
+        month,
+        year: month === 0 ? state.year + 1 : state.year,
+      };
+    });
   };
 
   render() {
-    const { minDate, maxDate, customInput, mode } = this.props;
-    const { date } = this.state;
+    const { isVisible, minDate, maxDate, onDismiss } = this.props;
+    const { date, month, year } = this.state;
 
-    const showTimeSelect = mode === 'time' || mode === 'datetime';
-    const showTimeSelectOnly = mode === 'time';
+    if (!isVisible) {
+      return null;
+    }
+
+    const monthLength = new Date(year, month + 1, 0).getDate();
+    const startDay = new Date(year, month, 1).getDay();
+    const selectedDay = new Date(
+      date.getFullYear(),
```

Comment:
I know that we recently had to use getUTCFullYear and the likes in `mobile` repo: https://github.com/kiwicom/mobile/commit/6d5e641de896e0b619775ea23f204a137d3a1a2a

Reply:
Thanks, I'll look at it.